rasa/shared/core/domain.py

```diff
@@ -278,9 +278,8 @@ def collect_slots(slot_dict: Dict[Text, Any]) -> List[Slot]:
         # make a copy to not alter the input dictionary
         slot_dict = copy.deepcopy(slot_dict)
         for slot_name in sorted(slot_dict):
-            slot_class = Slot.resolve_by_type(slot_dict[slot_name].get("type"))
-            if "type" in slot_dict[slot_name]:
-                del slot_dict[slot_name]["type"]
+            slot_class = Slot.resolve_by_type(slot_dict[slot_name].pop("type", None))
+
```

Comment:
why get rid of the key `type` deletion here?

Reply:
`pop` removes it in one go. I split it up in two lines as this is too nested